src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -299,7 +299,7 @@ else if (value instanceof ByteBuffer)
               }
             }
           } else {
-            objectMode = schema.keySchema().type() == Schema.Type.STRING;
+            objectMode = schema.keySchema() == null || schema.keySchema().type() == Schema.Type.STRING;
```

Comment:
what does it mean to have key schema as null here?

Reply:
The keySchema here, apparently, is sometimes null in our environment (reason unknown for now..). However, I've reverted this change for now, since it's unrelated to the intended bugfix #167 

Will investigate further and might make a new PR